src/node.rs

```diff
@@ -395,15 +404,27 @@ mod tests {
 
     #[test]
     fn set_measure() {
+        struct ReturnStoredSize {
+            size: f32,
+        }
+
+        impl Measure for ReturnStoredSize {
+            fn measure(&self, _node: Node, _size: Size<Option<f32>>) -> Size<f32> {
+                let ReturnStoredSize { size } = self;
+                Size { width: *size, height: *size }
+            }
+        }
+
         let mut taffy = Taffy::new();
-        let node = taffy
-            .new_leaf_with_measure(FlexboxLayout::default(), MeasureFunc::Raw(|_| Size { width: 200.0, height: 200.0 }))
-            .unwrap();
-        taffy.compute_layout(node, Size::undefined()).unwrap();
+        let node = taffy.new_leaf_with_required_measure(FlexboxLayout::default()).unwrap();
+
+        let mut measure = ReturnStoredSize { size: 200. };
+        taffy.compute_measured_layout(node, &measure, Size::undefined()).unwrap();
         assert_eq!(taffy.layout(node).unwrap().size.width, 200.0);
 
-        taffy.set_measure(node, Some(MeasureFunc::Raw(|_| Size { width: 100.0, height: 100.0 }))).unwrap();
-        taffy.compute_layout(node, Size::undefined()).unwrap();
+        measure.size = 100.;
+        taffy.mark_dirty(node).unwrap();
```

Comment:
`mark_dirty` outlines another issue, that data needed for measurements now completely decoupled from the API, thus it might be easy to forget to call `mark_dirty` on the node,